libs/wingc/src/type_check.rs

```diff
@@ -515,6 +515,21 @@ impl TypeRef {
 			false
 		}
 	}
+
+	pub fn is_capturable(&self) -> bool {
+		match **self {
+			Type::Resource(_) => true,
+			Type::Enum(_) => true,
+			Type::Number => true,
+			Type::String => true,
+			Type::Duration => true,
+			Type::Boolean => true,
+			Type::Array(v) => v.is_capturable(),
+			Type::Map(v) => v.is_capturable(),
+			Type::Set(v) => v.is_capturable(),
```

Comment:
`Type::Struct`?

Reply:
I can add that but can't really test it yet I think

Reply:
added